src/python/turicreate/test/test_sarray.py

```diff
@@ -252,6 +252,11 @@ def test_list_with_none_creation(self):
         self.assertEqual(len(g), len(tlist))
         for i in range(len(tlist)):
           self.assertEqual(g[i], tlist[i])
+    
+    def test_nan_to_none_type_conversion(self):
+        sa=SArray([1.0,2.0,float('nan'),3.0])
+        sa=sa.astype(int)
+        self.assertTrue(None in sa)
```

Comment:
This is good but it's better to to check that there is only one `None` and that it's where it's suppose to be. Also good to check the other elements are what we expect too. 

I would change this line to:
```python
expected = SArray([1, 2, None, 3])
for a, b in zip(sa, expected):
    self.assertEqual(a, b)
```